backends/kv/writer.go

```diff
@@ -685,3 +684,32 @@ func valueToTypedExpressionString(value interface{}) string {
 		return fmt.Sprintf("%v", value)
 	}
 }
+
+func valueToKeyString(value interface{}) string {
+	switch typedVal := value.(type) {
+	case string:
+		return typedVal
+	case time.Time:
+		seconds := typedVal.Unix()
+		nanos := typedVal.Nanosecond()
+		return fmt.Sprintf("%v:%v", seconds, nanos)
```

Comment:
Should probably remove this
```suggestion
```

Reply:
Why ? Looks like it fixes IG-22141 timestamps. Am I missing something?

Reply:
It's the wrong format – i.e. it's incompatible with storey, which writes timestamps as some version of ISO 8601, pandas that does roughly (but probably not exactly) the same, and spark that hopefully does what pandas does. The difficulty in nailing down a timestamp format is a strong argument in favour of blocking this feature (otherwise, we need to put in the time to standardize this across engines).

Reply:
See comments on [IG-22141](https://jira.iguazeng.com/browse/IG-22141) for discussion.